src/components/MessageList/MessageList.tsx

```diff
@@ -1,51 +1,175 @@
 `;
 
+export type MessageListProps<
+  At extends Record<string, unknown> = DefaultAttachmentType,
+  Ch extends Record<string, unknown> = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends Record<string, unknown> = DefaultEventType,
+  Me extends Record<string, unknown> = DefaultMessageType,
+  Re extends Record<string, unknown> = DefaultReactionType,
+  Us extends Record<string, unknown> = DefaultUserType
+> = {
+  /**
+   * Style object for action sheet (used to message actions).
+   * Supported styles: https://github.com/beefe/react-native-actionsheet/blob/master/lib/styles.js
+   */
+  actionSheetStyles?: { [key: string]: unknown };
+  /**
+   * Besides existing (default) UX behavior of underlying FlatList of MessageList component, if you want
+   * to attach some additional props to underlying FlatList, you can add it to following prop.
+   *
+   * You can find list of all the available FlatList props here - https://facebook.github.io/react-native/docs/flatlist#props
+   *
+   * **NOTE** Don't use `additionalFlatListProps` to get access to ref of flatlist. Use `setFlatListRef` instead.
+   *
+   * e.g.
+   * ```js
+   * <MessageList
+   *  additionalFlatListProps={{ bounces: true, keyboardDismissMode: true }} />
+   * ```
+   */
+  additionalFlatListProps?: FlatListProps<
+    InsertDate<At, Ch, Co, Ev, Me, Re, Us>
+  >;
+  /**
+   * Custom UI component for attachment icon for type 'file' attachment.
+   * Defaults to: https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/FileIcon.js
+   */
+  AttachmentFileIcon?: React.ComponentType<Partial<FileIconProps>>;
+  /**
+   * Date separator UI component to render
+   *
+   * Defaults to and accepts same props as: [DateSeparator](https://getstream.github.io/stream-chat-react-native/#dateseparator)
+   */
+  DateSeparator?: React.ComponentType<
+    DateSeparatorProps<At, Ch, Co, Ev, Me, Re, Us>
+  >;
+  disableWhileEditing?: boolean;
+  /**
+   * UI component for header of message list. By default message list doesn't have any header.
+   * This is basically a [ListFooterComponent](https://facebook.github.io/react-native/docs/flatlist#listheadercomponent) of FlatList
+   * used in MessageList. Its footer instead of header, since message list is inverted.
+   *
+   */
+  HeaderComponent?: React.ComponentType;
+  /**
+   * Custom UI component to display a message in MessageList component
+   * Default component (accepts the same props): [MessageSimple](https://getstream.github.io/stream-chat-react-native/#messagesimple)
+   */
+  Message?: React.ComponentType<Partial<unknown>>; // TODO: type with Message props
+  /**
+   * Array of allowed actions on message. e.g. ['edit', 'delete', 'reactions', 'reply']
+   * If all the actions need to be disabled, empty array or false should be provided as value of prop.
+   */
+  messageActions?: boolean | ('delete' | 'edit' | 'reactions' | 'reply')[];
```

Comment:
since this can be customized not sure we can do this, think it needs to be `boolean | string[]`

Reply:
word, is this a type within your Message component work? if so, I can change this and throw down a TODO to make it reference the type from Message when it's done

Reply:
since we allow people to pass whatever messageActions they want to the ActionSheet component, not sure it can be anything other than string[], that's how i was planning to do it